pandas/tests/frame/test_convert_to.py

```diff
@@ -175,7 +172,117 @@ def test_to_records_index_name(self):
     def test_to_records_with_unicode_index(self):
         # GH13172
         # unicode_literals conflict with to_records
-        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a')\
+        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a') \
             .to_records()
         expected = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
         tm.assert_almost_equal(result, expected)
+
+    def test_to_records_with_tz(self):
+        # GH13937
+        df_utc = DataFrame({'datetime': date_range('2016-01-01',
+                                                   periods=10,
+                                                   freq='S',
+                                                   tz='UTC')},
+                           index=date_range('2016-01-01',
+                                            periods=10,
+                                            freq='S',
+                                            tz='UTC'))
+
+        tm.assertIsNot(df_utc.to_records().index[0].tzinfo,
+                       df_utc.tz_convert("GMT").to_records().index[0].tzinfo)
+
+        tm.assert_numpy_array_equal(df_utc.to_records(),
+                                    df_utc.tz_convert("GMT").to_records())
+
+        # test DataFrame.to_records() with timezone conversion to UTC
+
+        df_gmt = DataFrame({'datetime': date_range('2016-01-01',
+                                                   periods=10,
+                                                   freq='S',
+                                                   tz='GMT')},
+                           index=date_range('2016-01-01',
+                                            periods=10,
+                                            freq='S',
+                                            tz='GMT'))
+
+        tm.assert_almost_equal(df_utc.to_records()['datetime'][0],
```

Comment:
what does this comparison add that the ``assert_numpy_array_equal`` does not? (which is the way we would like to compare these)